pytorch_lightning/utilities/apply_func.py

```diff
@@ -38,14 +39,43 @@ def apply_to_collection(data: Any, dtype: Union[type, tuple], function: Callable
     return data
 
 
+class TransferableDataType(ABC):
```

Comment:
in case someone has a suggestion for a better name, let me know :)

Reply:
cc: @williamFalcon